### ISSUE TITLE
Different Json type definitions for different ByteList fields 

### ISSUE THREAD

**Issue by Nashatyrev** [Consensys-Incorporated/teku#5623]
### Description

According to API schema definitions [here](https://ethereum.github.io/beacon-APIs) the `BeaconStateAltair.*_epoch_participation` fields should be represented as an array of bytes while the field `ExecutionPayload.extra_data` field should be represented as a hex byte string. While both fields have the same SSZ schema: `ByteList`

Currently that discrepancy is annihilated by the absence of the following 2 lines: 
```patch
Index: infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/schema/collections/SszPrimitiveListSchema.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/schema/collections/SszPrimitiveListSchema.java b/infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/schema/collections/SszPrimitiveListSchema.java
--- a/infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/schema/collections/SszPrimitiveListSchema.java	(revision 1440c4cc14007ac5b3ddc380722817b692b6c053)
+++ b/infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/schema/collections/SszPrimitiveListSchema.java	(date 1653912116533)
@@ -42,6 +42,8 @@
       return (SszPrimitiveListSchema<PrimT, SszPrimT, ?>) SszBitlistSchema.create(maxLength);
     } else if (elementSchema == SszPrimitiveSchemas.UINT64_SCHEMA) {
       return (SszPrimitiveListSchema<PrimT, SszPrimT, ?>) SszUInt64ListSchema.create(maxLength);
+    } else if (elementSchema == SszPrimitiveSchemas.BYTE_SCHEMA) {
+      return (SszPrimitiveListSchema<PrimT, SszPrimT, ?>) SszByteListSchema.create(maxLength);
     } else {
       return new SszPrimitiveListSchemaImpl<>(elementSchema, maxLength);
     }
```
In other words currently the `*_epoch_participation` schema is `SszPrimitiveListSchemaImpl<SszByte>` instance while `extra_data` schema is `SszByteListSchemaImpl` instance. That's why they have different Json type definitions. 

### Possible solutions

- Request API change: always represent `ByteList` as a hex string
- Request API change: always represent `ByteList` as an array of bytes
- Add a Json type parameter to the `SszByteListSchema` and specify different Json types for the fields mentioned above 

### Relates to
- #5611

**Comment by ajsutton** [comment]
This has been a bit of a point of contention recently - the participation lists in the state are defined as `List[uint8]` and the extra data is `ByteList`. In the ssz spec `byte` is an alias of `uint8` but the JSON representation of the two is different (one being a decimal number and one being hex encoded).  We should be able to use the same underlying representation in SSZ but with a different mapping to JSON easily enough though. I think it will boil down to just having a subclass of SszByteListSchema that provides the uint8 form of json type definition with no other changes. I'll have a play.
